[ROCm] Update rocm.yml and add rocm-mi300.yml

amdfaa opened the pull request:
- Added another workflow to run the mi300 jobs post-merge.
- Updated rocm.yml to use mi200s instead of mi300s.
- Required to get an idea of how PRs are landing on our mi200s and mi300s.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @naromero77amd

amdfaa committed:
Update rocm.yml

amdfaa committed:
Create rocm-mi300.yml

jeffdaily commented on .github/workflows/rocm-mi300.yml:
```diff
@@ -0,0 +1,59 @@
+name: rocm
```
I think this needs to be "rocm-mi300" so it shows up on the HUD uniquely.

jeffdaily committed:
updates

- don't use mi300 for trunk
- rename rocm-mi300
- don't use mi300 for release branches, only main

jeffdaily commented:
@pytorchbot merge -f "ROCm CI workflow changes, no code changes"
